testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1528,6 +1528,33 @@ public void testUpdateOriginalFilesUnpartitioned()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testRowIdCorrectness()
+    {
+        withTemporaryTable("test_delete_rowids", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format(
+                    "CREATE TABLE %s\n"
+                            + "WITH (transactional = true)\n"
+                            + "AS SELECT *\n"
+                            + "FROM tpch.sf1.lineitem\n"
```

Comment:
How much time does it take? We should use the smallest table which reproduces the issue. Maybe orders or even customers is enough?

Reply:
I briefly tried those, but they didn't work in my first few tries. Maybe it does work for some specific values, or if we can reduce the page size somehow?

Reply:
> or if we can reduce the page size somehow

I am not aware of any way to do that. It depends on how reader behaves.

Reply:
Inserting the data into this table seems to take just under 40 seconds.

Reply:
I'll poke around with the smaller tables a bit and see if I can get them to work.

Reply:
It turns out I'd made a silly mistake with my earlier testing, so I've changed the test to use `tpch.tiny.supplier`, which seems to be the smallest table that gets split into multiple pages.